wgsl/index.bs

```diff
@@ -9444,26 +9675,44 @@ See [[#statements]] and [[#function-calls]].
-concurrently on the GPU.  To ensure correct and portable behavior, WGSL
-requires that these operations can be statically analyzed to not have any
-control dependencies such that a non-empty strict subset of invocations will
-execute the operation (i.e. the operation must be executed in [=uniform control
-flow=]).
+A [[#collective-operations|collective operation]]
+(e.g. barrier, derivative, or a texture operation relying on an implicitly computed derivative)
+requires coordination among different invocations running concurrently on the GPU.
+The operation executes correctly and portably
+when all invocations execute it concurrently, i.e. in [=uniform control flow=].
+
+Conversely, incorrect or non-portable behavior occurs when a strict subset of invocations
+execute the operation, i.e. in non-uniform control flow.
+Informally, some invocations reach the collective operation, but others do not,
+or not at the same time, as a result of non-uniform control dependencies.
 Non-uniform control dependencies arise from [[#control-flow|control flow]]
 statements whose behavior depends on [=uniform value|non-uniform values=].
-These non-uniform values can be traced back to certain sources that are not
+
+> For example, a non-uniform control dependency arises when different invocations compute
+    different values for the condition of an
+    [=statement/if=], [=statement/break-if=], [=statement/while=], or [=statement/for=],
+    or different values for the selector of a [=statement/switch=].
+
+These non-uniform values can often be traced back to certain sources that are not
 statically proven to be uniform.
 These sources include, but are not limited to:
 * Mutable [=module scope|module-scope=] [=variables=]
 * Most [=built-in values=], except [=built-in values/num_workgroups=] and [=built-in values/workgroup_id=]
 * [=user-defined input datum|User-defined inputs=]
 * Certain [=built-in functions=] (see [[#uniformity-function-calls]])
 
-The remainder of this section is devoted to a description of this static
-analysis an implementation [=behavioral requirement|will=] perform to validate
-the WGSL program.
+To ensure correct and portable behavior, a WGSL implementation [=behavioral requirement|will=]
+perform a static <dfn>uniformity analysis</dfn>, attempting to prove that each collective operation
+executes in [=uniform control flow=].
+Subsequent subsections describe the analysis.
+
+A <dfn noexport>uniformity failure</dfn> [=behavioral requirement|will=] be triggered
+when [=uniformity analysis=] cannot prove that a particular [[#collective-operations|collective operation]] executes in [=uniform control flow=].
+* If a uniformity failure is triggered for a
+    [=builtin functions that compute a derivative|builtin function that computes a derivative=], then
+    a [=trigger/derivative_uniformity=] [=diagnostic=] is [=triggered=].
+    The [=diagnostic/triggering location=] for the diagnostic is the [=call site=] of that builtin.
```

Comment:
Is "call site" the right term to use here? The definition says:

> Call sites are a [dynamic context](http://raw.githack.com/gpuweb/gpuweb-pr-previews/e2ee9948ce503be489787d630290d5c9dbf2ae7c/gpuweb/gpuweb/3713/wgsl/index.html#dynamic-context). As such, the same textual location may represent multiple call sites.

This sounds to me like "call site" should not be used to refer to a specific textual location, but I'm not sure.

Reply:
Yes, good call.  Most times I think of call site as a text location, but you're right that it's defined that way.
Fixed this spot and also the spot at the definition of derivative_uniformity in 2.2.1 Filterable Triggering Rules.